### DISCUSSION TITLE
NLP: Beragam Bahasa & Morfologi Bahasa Indonesia

### DISCUSSION THREAD

**Discussion by Talioll** [teamalgoritma/community#483] [Q&A] [answered]
Selamat siang,

Ingin bertanya mengenai tahap text cleansing, kira-kira apa yang bisa dilakukan ketika menghadapi teks dengan beragam bahasa? Misalnya menggunakan bahasa daerah, bahasa inggris dan bahasa indonesia sekaligus dalam suatu teks?

Kemudian bagaimana menghadapi teks dengan morfologi bahasa indonesia? seperti kupu-kupu (tidak bisa diartikan sebagai kupu saja), atau rumah makan/restoran (jika dipisahkan per rumah dan makan akan menghasilkan arti yg berbeda pula). 

Demikian perntanyaan sy, mungkin ada penjelasan atau artikel yang dapat sy baca?

**Comment by nisa-basalamah** [comment]
Dear @Talioll,

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan referensi berkaitan dengan pertanyaan yang diberikan.

**Comment by idoali** [comment, marked as answer]
Selamat siang bu, terima kasih atas pertanyaannya

Untuk teks yang bahasanya tercampur seperti itu, mau tidak mau harus dipisahkan secara manual bu. 

Untuk kasus morfologi bahasa indonesia, ibu bisa mencoba menyelesaikan masalah tersebut dengan library `katadasaR`. Cara menyelesaikannya adalah dengan proses _stemming_, atau pemotongan imbuhan sehingga menjadi kata dasar. Untuk kasus unik seperti istilah "rumah makan", ini diperlukan treatment khusus. Caranya adalah dengan membuat fungsi baru.

Sebagai tambahan bu, deep learning sudah banyak digunakan di kasus pengolahan teks. Model yang paling ampuh digunakan untuk kasus pengolahan teks adalah Transformers. 

Transformers ini adalah model deep learning yang kompleks yang sudah di training dengan data yang sangat banyak. Contoh dari model transformers ini adalah ChatGPT. Ibu bisa menggunakan beberapa model transformers (GPT yang digunakan di ChatGPT itu hanya salah satu jenis dari sekian banyak model transformers) dengan R. Dengan transformers ini juga, kemungkinan Ibu bisa menyelesaikan masalah pengolahan teks yang ibu hadapi sekarang. 

Berikut referensinya jika ibu ingin tahu lebih lanjut.  

https://rpubs.com/eR_ic/transfoRmers

**Reply by Talioll** [reply]
Dear Mas @idoali,

Terima kasih atas jawabannya :)

Masukannya akan sy coba aplikasikan, dan referensinya juga akan sy baca. Terima kasih :D

**Reply by idoali** [reply]
Baik terima kasih kembali bu.

Ini juga ada tambahan bu untuk text cleaning bahasa indonesia https://algotech.netlify.app/blog/text-cleaning-bahasa/
